pam/internal/adapter/formmodel.go

```diff
@@ -24,12 +23,11 @@ func newFormModel(label, entryType, buttonLabel string, wait bool) formModel {
 	// TODO: add digits and force validation.
 	switch entryType {
 	case "chars":
-		entry := &textinputModel{Model: textinput.New()}
-		focusableModels = append(focusableModels, entry)
+		entry := newTextInputModel(entryType)
+		focusableModels = append(focusableModels, &entry)
 	case "chars_password":
-		entry := &textinputModel{Model: textinput.New()}
-		entry.EchoMode = textinput.EchoNone
-		focusableModels = append(focusableModels, entry)
+		entry := newTextInputModel(entryType)
+		focusableModels = append(focusableModels, &entry)
```

Comment:
case "chars" and case "chars_password" now do exactly the same, so we can simplify this:

```
	if entryType == "chars" || entryType == "chars_password" {
		entry := newTextInputModel(entryType)
		focusableModels = append(focusableModels, &entry)
	}
```

Reply:
oh, indeed that was left in my stash.